src/parser/scanner/regexp.ts

```diff
@@ -31,6 +31,19 @@ export function scanRegularExpression(parser: ParserState, source: string): Synt
           pos++;
           break loop;
         case Char.LeftBracket:
+          if (
+            context & Context.Lint &&
+            parser.linterFlags & LinterFlags.NonEmptyCharacterClass &&
+            source.charCodeAt(pos + 1) === Char.RightBracket
+          ) {
+            parser.onError(
+              DiagnosticSource.Lexer,
+              DiagnosticKind.Error,
+              diagnosticMap[DiagnosticCode.Empty_class],
+              parser.curPos,
+              pos
+            );
+          }
           preparseState |= Preparse.InCharClass;
```

Comment:
@aladdin-add If this is correct. you merge!

Reply:
@aladdin-add  In the future we could skip the  context & `Context.Lint &&` and check directly with `parser.linterFlags &`. It speeds up the linting with one less check, but I haven't prioritized linting. And some of this "rules" need to be grouped and separated into different bitwise sets.

Is this something you can fix?